app/forms/node_form.rb

```diff
@@ -49,8 +53,13 @@ def self.node_permitted(node, permission_type, groups)
 
   def create_objects
     ActiveRecord::Base.transaction do
-      node = Node.create!(name: name, cost_code: cost_code, description: description, parent_id: parent_id, owner: @owner)
-      node.permissions.create!(convert_permissions(@owner))
+      node = Node.new(name: name, cost_code: cost_code, description: description, parent_id: parent_id, owner: @owner)
+      if node.save
+        node.permissions.create!(convert_permissions(@owner))
+      else
+        @node = node
```

Comment:
This is very side affecty. You could just set the initial `node` as an instance variable i.e. `@node = Node.new(...)`